waterlog/src/actions/LeakCostsActions.js

```diff
@@ -0,0 +1,15 @@
+import { FETCH_LEAKS_COSTS } from "./types";
+
+export const fetchLeaksCosts = (id = 1) => dispatch => {
+  fetch(`https://localhost:44382/api/segmentleaks/costs/${id}`)
+    .then(res => res.json())
+    .then(costs => {
+      dispatch({
+        type: FETCH_LEAKS_COSTS,
+        payload: costs
+      });
+    })
+    .catch(function (ex) {
+      throw ex;
```

Comment:
You are catching the error and throwing it again. Are you catching it again where you call `fetchLeaksCosts`? It would make more sense to dispatch an action and handle the error in the reducer.

Reply:
I have dispatched an action to catch errors